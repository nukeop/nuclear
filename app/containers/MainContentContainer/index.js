import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as Actions from '../../actions';

import MainLayout from '../../components/MainLayout';

import AlbumViewContainer from '../AlbumViewContainer';
import ArtistViewContainer from '../ArtistViewContainer';
import DashboardContainer from '../DashboardContainer';
import FavoritesContainer from '../FavoritesContainer';
import LyricsContainer from '../LyricsContainer';
import PlaylistsContainer from '../PlaylistsContainer';
import PlaylistViewContainer from '../PlaylistViewContainer';
import PluginsContainer from '../PluginsContainer';
import SearchResultsContainer from '../SearchResultsContainer';
import SettingsContainer from '../SettingsContainer';
import TagViewContainer from '../TagViewContainer';
import EqualizerViewContainer from '../EqualizerViewContainer';
import LibraryViewContainer from '../LibraryViewContainer';

import DownloadsContainer from '../../containers/DownloadsContainer';

class MainContentContainer extends React.Component {
  componentDidMount () {
    if (this.props.history &&
      this.props.location &&
      this.props.location.pathname === '/') {
      this.props.history.push('/dashboard');
    }
  }

  render () {
    return (
      <Route render={({ location }) => {
        return (
          <MainLayout>
            <Switch key={location.key} location={location}>
              <Route path='/album/:albumId' component={AlbumViewContainer} />
              <Route path='/artist/:artistId' component={ArtistViewContainer} />
              <Route path='/dashboard' component={DashboardContainer} />
              <Route path='/downloads' component={DownloadsContainer} />
              <Route path='/favorites/tracks' component={FavoritesContainer} />
              <Route path='/playlists' component={PlaylistsContainer} />
              <Route path='/playlist/:playlistId' component={PlaylistViewContainer} />
              <Route path='/plugins' component={PluginsContainer} />
              <Route path='/settings' component={SettingsContainer} />
              <Route path='/tag/:tagName' component={TagViewContainer} />
              <Route path='/search' component={SearchResultsContainer} />
              <Route path='/lyrics' component={LyricsContainer} />
              <Route path='/equalizer' component={EqualizerViewContainer} />
              <Route path='/library' component={LibraryViewContainer} />
            </Switch>
          </MainLayout>
        );
      }
      } />
    );
  }
}

function mapStateToProps () {
  return {};
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators(Actions, dispatch)
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MainContentContainer));
