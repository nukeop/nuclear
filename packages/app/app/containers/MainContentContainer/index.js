import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as SearchActions from '../../actions/search';

import MainLayout from '../../components/MainLayout';

import AlbumViewContainer from '../AlbumViewContainer';
import ArtistViewContainer from '../ArtistViewContainer';
import DashboardContainer from '../DashboardContainer';
import DownloadsContainer from '../DownloadsContainer';
import EqualizerViewContainer from '../EqualizerViewContainer';
import FavoritesContainer from '../FavoritesContainer';
import LibraryViewContainer from '../LibraryViewContainer';
import LyricsContainer from '../LyricsContainer';
import PlaylistViewContainer from '../PlaylistViewContainer';
import PlaylistsContainer from '../PlaylistsContainer';
import PluginsContainer from '../PluginsContainer';
import SearchResultsContainer from '../SearchResultsContainer';
import SettingsContainer from '../SettingsContainer';
import TagViewContainer from '../TagViewContainer';
import VisualizerNode from '../VisualizerContainer/VisualizerNode';

class MainContentContainer extends React.Component {
  componentDidMount () {
    if (this.props.history &&
      this.props.location &&
      this.props.location.pathname === '/') {
      this.props.history.push('/dashboard');
    }
  }

  render () {
    return (
      <Route render={({ location }) => {
        return (
          <MainLayout>
            <Switch key={location.key} location={location}>
              <Route path='/album/:albumId' component={AlbumViewContainer} />
              <Route path='/artist/:artistId' component={ArtistViewContainer} />
              <Route path='/dashboard' component={DashboardContainer} />
              <Route path='/downloads' component={DownloadsContainer} />
              <Route path='/favorites/albums' component={FavoritesContainer} />
              <Route path='/favorites/tracks' component={FavoritesContainer} />
              <Route path='/favorites/artists' component={FavoritesContainer} />
              <Route path='/playlists' component={PlaylistsContainer} />
              <Route path='/playlist/:playlistId' component={PlaylistViewContainer} />
              <Route path='/plugins' component={PluginsContainer} />
              <Route path='/settings' component={SettingsContainer} />
              <Route path='/tag/:tagName' component={TagViewContainer} />
              <Route path='/search' component={SearchResultsContainer} />
              <Route path='/lyrics' component={LyricsContainer} />
              <Route path='/equalizer' component={EqualizerViewContainer} />
              <Route path='/visualizer' component={VisualizerNode} />
              <Route path='/library' component={LibraryViewContainer} />
            </Switch>
          </MainLayout>
        );
      }
      } />
    );
  }
}

function mapStateToProps () {
  return {};
}

function mapDispatchToProps (dispatch) {
  return {
    searchActions: bindActionCreators(SearchActions, dispatch)
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MainContentContainer));
