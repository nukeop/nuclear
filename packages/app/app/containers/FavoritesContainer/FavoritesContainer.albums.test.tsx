
import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import { buildStoreState } from '../../../test/storeBuilders';
import { AnyProps, configureMockStore, setupI18Next, TestRouterProvider, TestStoreProvider } from '../../../test/testUtils';
import MainContentContainer from '../MainContentContainer';

const updateStore = (favorites: object) => {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const store = require('@nuclear/core').store;
  store.set('favorites', favorites);
};

describe('Album view container', () => {
  beforeAll(() => {
    setupI18Next();
  });

  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const { store } = require('@nuclear/core');
    store.clear();
  });

  it('should display favorite albums', () => {
    const favorites = buildStoreState()
      .withFavorites()
      .build()
      .favorites;
    updateStore(favorites);
    const { component } = mountComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should display empty state', () => {
    const { component } = mountComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should go to album page after clicking an album', async () => {
    const favorites = buildStoreState()
      .withFavorites()
      .build()
      .favorites;
    updateStore(favorites);
    const { component, history } = mountComponent();

    await waitFor(() => component.getByText(/test album/i).click());

    expect(history.location.pathname).toBe('/album/test-album-1');
  });

  const mountComponent = (initialStore?: AnyProps) => {
    const initialState = initialStore ||
      buildStoreState()
        .build();

    const history = createMemoryHistory({
      initialEntries: ['/favorites/albums']
    });
    const store = configureMockStore(initialState);
    const component = render(
      <TestRouterProvider
        history={history}
      >
        <TestStoreProvider
          store={store}
        >
          <MainContentContainer />
        </TestStoreProvider>
      </TestRouterProvider >
    );
    return { component, history, store };
  };
});
