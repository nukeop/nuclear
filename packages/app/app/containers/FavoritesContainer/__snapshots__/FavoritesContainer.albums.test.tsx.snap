// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album view container should display empty state 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="favorite_albums_view"
    >
      <div
        class="empty_state"
      >
        <i
          class="icons"
        >
          <i
            aria-hidden="true"
            class="star icon"
          />
          <i
            aria-hidden="true"
            class="dot circle corner icon"
          />
        </i>
        <h2>
          No favorite albums
        </h2>
        <div>
          Try adding some albums to favorites and they will appear here!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Album view container should display favorite albums 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="favorite_albums_view"
    >
      <div
        class="header_container"
      >
        Your favorite albums
      </div>
      <div
        class="ui segment favorite_albums_segment"
      >
        <div
          class="nuclear album_grid auto_size"
        >
          <div
            class="album_cards"
          >
            <div
              class="nuclear card_container"
            >
              <div
                class="card animated"
              >
                <div
                  aria-expanded="false"
                  class="ui basic dropdown menu_button"
                  role="listbox"
                  tabindex="0"
                >
                  <i
                    aria-hidden="true"
                    class="ellipsis vertical icon"
                  />
                  <div
                    class="menu transition"
                  >
                    <div
                      class="item"
                      role="option"
                    >
                      Remove
                    </div>
                  </div>
                </div>
                <div
                  class="thumbnail"
                >
                  <div
                    class="overlay"
                  />
                </div>
                <div
                  class="card_content"
                >
                  <h4>
                     Test album 1
                  </h4>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
